cmd/lotus-miner/actor.go

```diff
@@ -530,16 +531,25 @@ var actorControlList = &cli.Command{
 		}
 
 		printKey := func(name string, a address.Address) {
-			b, err := api.WalletBalance(ctx, a)
-			if err != nil {
-				fmt.Printf("%s\t%s: error getting balance: %s\n", name, a, err)
+			var actor *types.Actor
+			if actor, err = api.StateGetActor(ctx, a, types.EmptyTSK); err != nil {
+				fmt.Printf("%s\t%s: error getting actor: %s\n", name, a, err)
 				return
 			}
-
-			k, err := api.StateAccountKey(ctx, a, types.EmptyTSK)
-			if err != nil {
-				fmt.Printf("%s\t%s: error getting account key: %s\n", name, a, err)
-				return
+			b := actor.Balance
+
+			var k address.Address
+			// 'a' maybe a 'robust', in that case, 'StateAccountKey' returns an error.
+			if builtin2.IsAccountActor(actor.Code) {
+				if k, err = api.StateAccountKey(ctx, a, types.EmptyTSK); err != nil {
+					fmt.Printf("%s\t%s: error getting account key: %s\n", name, a, err)
+					return
+				}
+			} else { // if builtin2.IsMultisigActor(actor.Code), Are there any other cases exist?
```

Comment:
With FVM you'll be able to use custom actors to proxy/filter calls to the miner actor, so in not-so-distant future we'll see non-account/non-multisig actors here